File: iOS/DoseNet/DoseNet/Dosimeter.swift
//
//  Dosimeter.swift
//  DoseNet
//
//  Created by Navrit Bal on 06/01/2016.
//  Copyright © 2016 navrit. All rights reserved.
//

import Foundation
import MapKit

class Dosimeter {
    let name : String!
    let lastDose_uSv : Double!
    let lastDose_mRem : Double!
    let lastTime : String!
    let distance : Double?
    let CLLat : CLLocation?
    let CLLon : CLLocation?
    
    init(name:String, lastDose_uSv: Double, distance: Double) {
            self.name = name
            self.lastDose_uSv = lastDose_uSv
            self.lastDose_mRem = nil
            self.lastTime = nil
            self.distance = distance
            self.CLLat = nil
            self.CLLon = nil
    }
    
    init(name:String, lastDose_uSv: Double, lastDose_mRem: Double,
        lastTime: String) {
            self.name = name
            self.lastDose_uSv = lastDose_uSv
            self.lastDose_mRem = lastDose_mRem
            self.lastTime = lastTime
            self.distance = nil
            self.CLLat = nil
            self.CLLon = nil
    }
    
    init(name:String, lastDose_uSv: Double, lastDose_mRem: Double,
        lastTime: String, distance: Double) {
            self.name = name
            self.lastDose_uSv = lastDose_uSv
            self.lastDose_mRem = lastDose_mRem
            self.lastTime = lastTime
            self.distance = distance
            self.CLLat = nil
            self.CLLon = nil
    }
    
    init(name:String, lastDose_uSv: Double, lastDose_mRem: Double,
        lastTime: String, distance: Double, CLLat:CLLocation, CLLon:CLLocation) {
            self.name = name
            self.lastDose_uSv = lastDose_uSv
            self.lastDose_mRem = lastDose_mRem
            self.lastTime = lastTime
            self.distance = distance
            self.CLLat = CLLat
            self.CLLon = CLLon
    }
    
}

File: Android/DoseNet/app/src/main/java/com/navrit/dosenet/Dosimeter.java
package com.navrit.dosenet;

public class Dosimeter {
    String name;
    double lastDose_uSv;
    double lastDose_mRem;
    String unitSelected;
    String lastTime;
    //double distance;
    double lat;
    double lon;

    // Image (seal/crest) for card view

    //List<double> doses_uSv; // For dose over time plots
    //List<String> times;

    public Dosimeter(String name, double lastDose_uSv, String unitSelected, String lastTime){
        this.name = name;
        this.lastDose_uSv = lastDose_uSv;
        this.lastDose_mRem = lastDose_uSv/10;
        this.unitSelected = unitSelected;
        this.lastTime = lastTime;
    }

    public Dosimeter(String name, double lastDose_uSv, String unitSelected, String lastTime,
                     double lon, double lat){
        this.name = name;
        this.lastDose_uSv = lastDose_uSv;
        this.lastDose_mRem = lastDose_uSv/10;
        this.unitSelected = unitSelected;
        this.lastTime = lastTime;
        this.lon = lon;
        this.lat = lat;
    }

}
